File: nodejs-ainame/src/routes/order/product.ts
import { Router, Request, Response } from 'express';
import { z } from 'zod';
import { prisma } from '../../lib/prisma';

const router = Router();

const ProductIn = z.object({
  name: z.string().max(100),
  price: z.number(),
  storenum: z.number(),
  description: z.string().max(500).optional(),
  productno: z.string().max(50).optional(),
});
const ProductUpdateIn = z.object({
  id: z.number(),
  name: z.string().max(100).optional(),
  price: z.number().optional(),
  storenum: z.number().optional(),
  description: z.string().max(500).optional(),
  productno: z.string().max(50).optional(),
});

/** POST /product/create - 创建产品 */
router.post('/create', async (req: Request, res: Response) => {
  const parsed = ProductIn.safeParse(req.body);
  if (!parsed.success) {
    res.status(400).json({ detail: parsed.error.message });
    return;
  }
  try {
    await prisma.product.create({
      data: {
        name: parsed.data.name,
        price: parsed.data.price,
        storenum: parsed.data.storenum,
        description: parsed.data.description ?? '',
        productno: parsed.data.productno ?? '',
      },
    });
    res.json({ code: 200, message: '创建成功', data: 'created successfully' });
  } catch (e) {
    console.error(e);
    res.status(500).json({ detail: String(e) });
  }
});

/** GET /product/query - 查询产品列表 */
router.get('/query', async (_req: Request, res: Response) => {
  try {
    const list = await prisma.product.findMany({ orderBy: { id: 'asc' } });
    res.json({ code: 200, message: '查询成功', data: list });
  } catch (e) {
    console.error(e);
    res.status(500).json({ detail: String(e) });
  }
});

/** PUT /product/update - 更新产品 */
router.put('/update', async (req: Request, res: Response) => {
  const parsed = ProductUpdateIn.safeParse(req.body);
  if (!parsed.success) {
    res.status(400).json({ detail: '参数错误' });
    return;
  }
  const { id, ...data } = parsed.data;
  try {
    await prisma.product.update({ where: { id }, data });
    res.json({ code: 200, message: '更新成功' });
  } catch (e) {
    res.status(404).json({ detail: '产品未找到' });
  }
});

/** DELETE /product/delete?prduct_id= (保持与原 API 拼写一致) */
router.delete('/delete', async (req: Request, res: Response) => {
  const id = parseInt((req.query.prduct_id || req.query.product_id) as string, 10);
  if (isNaN(id)) {
    res.status(400).json({ detail: '缺少 prduct_id 或 product_id' });
    return;
  }
  try {
    await prisma.product.delete({ where: { id } });
    res.json({ code: 200, message: '删除成功' });
  } catch (e) {
    res.status(404).json({ detail: '产品未找到' });
  }
});

export default router;

File: zhiliao-ainame/routers/order/product.py
from fastapi import APIRouter, Depends, HTTPException, Path, Query
from models import AsyncSession
from dependencies import get_session
from repository.order.product_repo import ProductRepository
from schemas.art import ArtQueryOut
from typing import Optional, Annotated
from pydantic import BaseModel, Field


# 创建公司相关的路由组
router = APIRouter(prefix="/product", tags=["product"])

class ProductIn(BaseModel):
    """
    产品创建请求模型
    
    Attributes:
        name (str): 校名，最大长度100字符
        address (str): 学校地址，最大长度200字符
    """
    id: Annotated[
        int | None,
        Field(description="产品ID")
    ]
    name: Annotated[
        str,
        Field(..., max_length=100, description="校名")
    ]
    price: Annotated[
        float,
        Field(..., description="产品价格")
    ]
    storenum: Annotated[
        int,
        Field(..., description="库存数量")
    ]
    description: Annotated[
        str,
        Field(..., max_length=200, description="产品描述")
    ]
    productno: Annotated[
        str,
        Field(..., max_length=200, description="产品编号")
    ]

@router.post("/create", response_model=ArtQueryOut, summary="创建产品", description="创建一个新产品")
async def create_product(
    data: ProductIn,
    session: AsyncSession = Depends(get_session)
):
    try:
        repo = ProductRepository(session=session)
        company = await repo.create(data.name, data.price, data.storenum, data.description, data.productno)
        await session.commit()
        return ArtQueryOut(data='created successfully')
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.get("/query", summary="查询产品", description="根据ID查询产品信息")
async def query_product(
    session: AsyncSession = Depends(get_session)
):
    try:
        repo = ProductRepository(session=session)
        product = await repo.query()
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
    # await session.commit()
    # return ArtQueryOut(data=product) # 坑哈，这个类里包含了ArtOut,所以报错了
    return {
        "code": 200,
        "message": "查询成功",
        "data": product
    }

@router.put("/update", summary="更新产品", description="根据ID更新产品信息")
async def update_product(
    data: ProductIn,
    session: AsyncSession = Depends(get_session)
):
    try:
        repo = ProductRepository(session=session)
        result = await repo.update(data.id, data.name, data.price, data.storenum, data.description, data.productno)
        await session.commit()
        return {
            "code": 200,
            "message": "更新成功",
            "data": result
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
    
@router.delete("/delete", summary="删除公司", description="根据ID删除公司信息")
async def delete_company(
    prduct_id: int = Query(..., description="公司ID"),
    session: AsyncSession = Depends(get_session)
):
    try:
        repo = ProductRepository(session)
        result = await repo.delete(prduct_id)
        await session.commit()
        return {
            "code": 200,
            "message": "删除成功",
            "data": result
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))